server/controllers/order.js

```diff
@@ -20,22 +20,44 @@ const newOrderValidation = (order) => {
   }
 };
 
-const getAllOrders = (req, res) => {
-  const ordersList = orderModel.getAllOrders();
-  logger.info('Orders were fetched', ordersList);
+const userPermission = (username, orderUser) => {
+  if (username !== orderUser) {
+    throw new Error('Permission denied!');
+  }
+};
+
+const getAllOrders = async (req, res) => {
+  let foundOrders;
+  try {
+    foundOrders = await Order.find().exec();
+  } catch (err) {
+    logger.error(err);
+    return res.status(500).json({
+      message: err,
+    });
+  }
+
+  if (!foundOrders) {
+    return res.status(404).json({
+      message: 'There are no orders to fetch',
+    });
+  }
+  logger.info('Fetching all orders');
   return res.status(200).json({
-    message: 'Orders were fetched',
-    ordersList,
+    message: 'Orders details',
+    foundOrders,
   });
 };
 
 const createOrder = (req, res) => {
   const { body } = req;
-  const order = {
+  const order = new Order({
     userName: body.userName,
     product: body.product,
     date: Date.now(),
-  };
+  });
+
+  order.save();
```

Comment:
you created an order before validating, and you didn't handle it, it is a promise, either use then and catch or use await with try and catch